pyurtrcon.py

```diff
@@ -1,119 +1,164 @@
+        except Exception as e:
+            print('Error! Details: {}'.format(e.message))
+
+    def complete_rcon(self, primitive, *_):
+        if primitive:
+            return [ command for command in self.common_commands if command.startswith(primitive) ]
+        else:
+            return self.common_addresses
+
+    def do_history(self, _):
+        """Prints all the previous commands of this session."""
+        for entry in self._hist:
+            print(entry)
+
+    def do_help(self, args):
+        """'help' or '?' with no arguments will print all commands.\n'help <command>' or '? <command>' will print help on <command>."""
+        cmd.Cmd.do_help(self, args)
+
+    def do_exit(self, _):
+        """Exits from the PyUrTRCon tool."""
+        return -1
+
+    def do_EOF(self, _):
+        return -1
+
+    def preloop(self):
+        self._hist  = []
+        self._locals  = {}
+        self._globals = {}
+        cmd.Cmd.preloop(self)
+
+    def postloop(self):
+        cmd.Cmd.postloop(self)
+
+    def precmd(self, line):
+        self._hist += [ line.strip() ]
+        return line
+
+    def postcmd(self, stop, line):
+        return stop
+
+    def emptyline(self):   
+        pass
+
+    def default(self, line):      
+        print('Unrecognized command. Type "help" for instructions.')
+
+    undoc_header = None
+
+    def print_topics(self, header, cmds, cmdlen, maxcol):
+        if header is not None:
+            cmd.Cmd.print_topics(self, header, cmds, cmdlen, maxcol)
+
+
+class GistAuth(object):
+    gist_api = 'https://api.github.com/gists'
+
+    def __init__(self, gist_id, enc_key):
+        self.gist_id = gist_id
+        self.enc_key = enc_key
+
+    def _load_gist_content(self):
+        gist = urllib2.urlopen('{}/{}'.format(self.gist_api, self.gist_id))
+        gist_data = json.load(gist)
+        return gist_data.get('files').get('RconCipherData').get('content')
+
+    def _decrypt_password(self, rcon_cipher):
+        try:
+            from Crypto.Cipher import AES
+
+            data = base64.b64decode(rcon_cipher)
+            iv = data[:16]
+            cipher = AES.new(self.enc_key, AES.MODE_CBC, iv)
+            return (lambda x: x[:-ord(x[(len(x) - 1):])])(cipher.decrypt(data[16:]))
+
+        except ImportError:
+            print('(!) Install the PyCrypto library for GistAuth to work.')
+            return ''
+
+    def get_password(self):
+        pw_cipher = self._load_gist_content()
+        pw_raw = self._decrypt_password(pw_cipher)
+        return pw_raw
+
+
+def main():
+    console = Console()
+    if len(sys.argv) >= 2:
+        console.server_host = sys.argv[1].split(':')[0] if (':' in sys.argv[1]) else sys.argv[1]
+        console.server_port = int(sys.argv[1].split(':')[1]) if (':' in sys.argv[1]) else 27960
+
+    gistauth_data = os.environ.get('PYURTRCON_GISTAUTH')
+
+    if gistauth_data is not None:
+        server, gist_id, enc_key = gistauth_data.split('|')
```

Comment:
Yes, you're right. I just thought it would be too much to ask of the admins to put all these stuff in their environment.

Nevertheless, I think I'll be discontinuing this script and start an ionic webapp.